src/pageNavigation.js

```diff
@@ -148,7 +147,12 @@ async function addListener(listener, {
 
     // Compile the match patterns for the listener
     const matchPatternSet = matching.createMatchPatternSet(matchPatterns);
-    const contentScript = await contentScripts.registerContentScript(matchPatterns, pageNavigationContentScript);
+    const contentScript = await browser.scripting.registerContentScripts([{
```

Comment:
Actually, on that note, this new function returns an array instead of a single object. Should this be `const [contentScript] = ...` instead?

Reply:
We are storing it in the `pageDataListeners` below. I'm not sure if that's used anywhere, but seems wise to be consistent for posterity

Reply:
Yeah, I don't think this is really working as intended, but I'll go ahead and change it anyway :) I'd like to make it possible to unload all of these content scripts but it doesn't look like everything is wired up correctly right now.